nautobot/extras/utils.py

```diff
@@ -426,21 +417,50 @@ def refresh_job_model_from_job_class(job_model_class, job_source, job_class, *,
             JOB_MAX_NAME_LENGTH,
         )
 
+    # handle duplicate names by appending an incrementing counter to the end
+    default_job_name = job_class.name[:JOB_MAX_NAME_LENGTH]
+    job_name = default_job_name
+    append_counter = 2
+    while (
+        job_model_class.objects.filter(name=job_name)
+        .exclude(
+            source=job_source[:JOB_MAX_SOURCE_LENGTH],
+            git_repository=git_repository,
+            module_name=job_class.__module__[:JOB_MAX_NAME_LENGTH],
+            job_class_name=job_class.__name__[:JOB_MAX_NAME_LENGTH],
+        )
+        .exists()
+    ):
```

Comment:
Can we simplify this to this perhaps?
```suggestion
    while (len(job_model_class.objects.filter(name=job_name)) > 1):
```